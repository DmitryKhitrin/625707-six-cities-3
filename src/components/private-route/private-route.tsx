import React, {FC} from 'react';
import {Route, Redirect} from 'react-router-dom';

type Props = {
  path: string;
  require: boolean;
  to: string;
  render: () => any;
}

export const PrivateRoute: FC<Props> = ({render, path, require, to}) => {
  return <Route path={path} exact render={() => require ? render() : <Redirect to={to} /> } />;
};

