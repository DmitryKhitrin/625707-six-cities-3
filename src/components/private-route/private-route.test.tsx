import React, {FC} from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {PrivateRoute} from './private-route';
import {BrowserRouter} from 'react-router-dom';

Enzyme.configure({
  adapter: new Adapter()
});

const TestComponent: FC = () => (<div>Hello</div>);

describe(`<PrivateRoute />`, () => {

  it(`should render Component.`, () => {

    const tree = Enzyme.mount(
        <BrowserRouter>
          <PrivateRoute
            path="/"
            to="/dublin"
            render={() => <TestComponent />}
            require={true}
          />
        </BrowserRouter>
    );

    expect(tree.getDOMNode()).toMatchSnapshot();
  });

  it(`should render null.`, () => {
    const tree = Enzyme.mount(
        <BrowserRouter>
          <PrivateRoute
            path="/"
            to="/dublin"
            render={() => <TestComponent />}
            require={false}
          />
        </BrowserRouter>
    );

    expect(tree.getDOMNode()).toMatchSnapshot();
  });
});
