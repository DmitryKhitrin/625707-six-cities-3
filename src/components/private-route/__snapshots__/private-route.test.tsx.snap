// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should render Component. 1`] = `ReactWrapper {}`;

exports[`<PrivateRoute /> should render null. 1`] = `ReactWrapper {}`;
